lib/rules/no-arrow-function-lifecycle.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @fileoverview It is not necessary to use arrow function for lifecycle methods
+ * @author Tan Nguyen
+ */
+'use strict';
+
+const Components = require('../util/Components');
+const astUtil = require('../util/ast');
+const docsUrl = require('../util/docsUrl');
+
+const lifecycleMethods = [
+  'getDefaultProps',
+  'getInitialState',
+  'getChildContext',
+  'getDerivedStateFromProps',
+  'componentWillMount',
+  'UNSAFE_componentWillMount',
+  'componentDidMount',
+  'componentWillReceiveProps',
+  'UNSAFE_componentWillReceiveProps',
+  'shouldComponentUpdate',
+  'componentWillUpdate',
+  'UNSAFE_componentWillUpdate',
+  'getSnapshotBeforeUpdate',
+  'componentDidUpdate',
+  'componentDidCatch',
+  'componentWillUnmount',
+  'render'
+];
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'It is not necessary to use arrow function for lifecycle methods',
+      category: 'Best Practices',
+      recommended: true,
+      url: docsUrl('no-arrow-function-lifecycle')
+    },
+    schema: []
+  },
+
+  create: Components.detect((context, components) => {
+    /**
+     * @param {Array} properties list of component properties
+     */
+    function reportNoArrowFunctionLifecycle(properties) {
+      properties.forEach(node => {
+        const propertyName = astUtil.getPropertyName(node);
+        const nodeType = node.value && node.value.type;
+        const isLifecycleMethod = lifecycleMethods.indexOf(propertyName) !== -1;
+
+        if (nodeType === 'ArrowFunctionExpression' && isLifecycleMethod) {
+          context.report({
+            node,
+            message: '{{propertyName}} is unnecessary to use arrow function',
```

Comment:
let's have this message be more like `{{propertyName}} is a React lifecycle method, and should not be an arrow function. Use an instance method instead.`